src/CodeReview.Orchestrator/Services/ContainerLogListener.cs

```diff
@@ -0,0 +1,88 @@
+﻿using System;
+using System.Buffers;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Docker.DotNet;
+using Microsoft.Extensions.Logging;
+
+namespace GodelTech.CodeReview.Orchestrator.Services
+{
+    public sealed class ContainerLogListener : IContainerLogListener
+    {
+        private readonly ILogger<ContainerLogListener> _logger;
+
+        private readonly string _containerId;
+        private readonly MultiplexedStream _stream;
+        private readonly CancellationTokenSource _cancellationTokenSource;
+
+        private Task _listenTask;
+
+        public ContainerLogListener(ILogger<ContainerLogListener> logger,
+            long waitTimeoutSeconds,
+            string containerId, 
+            MultiplexedStream stream)
+        {
+            _logger = logger ?? throw new ArgumentNullException(nameof(logger));
+            _containerId = containerId ?? throw new ArgumentNullException(nameof(containerId));
+            _stream = stream ?? throw new ArgumentNullException(nameof(stream));
+
+            _cancellationTokenSource = new CancellationTokenSource(TimeSpan.FromSeconds(waitTimeoutSeconds));
+        }
+
+        public void StartListening()
+        {
+            if (_listenTask != null)
+                throw new InvalidOperationException("Already listening a stream");
+
+            _listenTask = Task.Factory.StartNew(Listen, _cancellationTokenSource.Token, TaskCreationOptions.LongRunning);
+        }
+
+        public void StopListening()
+        {
+            if (_listenTask == null || _cancellationTokenSource.IsCancellationRequested)
+                return;
+
+            _cancellationTokenSource.Cancel();
+        }
+
+        private void Listen(object obj)
+        {
+            var buffer = ArrayPool<byte>.Shared.Rent(81920);
```

Comment:
Source text
`81920` number looks strange to me. To my mind it's better to introduce constant to explain this number (even though it's used only once)